pkg/cache/config.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package cache
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+type cacheType string
+
+const (
+	timeDay              = 24 * time.Hour
+	defaultCacheValidity = 5 * timeDay
+	cacheTypeEmpty       = cacheType("")
+	cacheTypeNone        = cacheType("none")
+	cacheTypeSQLite      = cacheType("sqlite")
+)
+
+// Config holds the cache configuration.
+type Config struct {
+	// Type is the type of the cache.
+	Type cacheType
+	// Validity is the duration for which the cache is valid.
+	Validity time.Duration
+	// Jitter is the jitter to apply when considering a cached entry valid or not.
+	Jitter time.Duration
+
+	cacheParser *configParser
+}
+
+// NewConfig is the constructor for Config.
+func NewConfig() Config {
+	return Config{
+		cacheParser: newConfigParser(),
+	}
+}
+
+// Present tell whether a cache configuration is present.
+func (c *Config) Present() bool {
```

Comment:
"present" 
<img src="https://media3.giphy.com/media/giKdkKxdDWecg524cE/giphy.gif"/>

(: 

What about: 
```suggestion
func (c *Config) IsSet() bool {
```